### DISCUSSION TITLE
Virt:  Have a way to change the random seed

### DISCUSSION THREAD

**Discussion by sosthene-nitrokey** [trussed-dev/trussed#46] [Ideas]
Right now the random seed is always set to `[42u8;32]`, which is fine for tests. It might be worth have a way to change it, for example in fuzzing it would be desirable to not generate always the same keys, and extract the seed from the fuzz input.

**Comment by robin-nitrokey** [comment]
Probably, the default should be a random seed, providing the option to override it with a static seed for reproducibility tests or a seed from a different source of randomness.  Unfortunately I don’t know what regression is meant by this comment: https://github.com/trussed-dev/trussed/blob/baab96d27e884c0133f1cc021d37ad93b8f2ecf8/src/virt.rs#L38-L39

Suggestion:
1. Introduce `struct Options { seed: Option<...>, }`.
2. Add `with_platform_options` and `with_client_options` and use `Options::default()` in the existing methods.
3. If no seed is set in the options, generate it randomly.  Otherwise, use the specified seed.
